### PR TITLE
[opentelemetry-operator] - BUMP to version 0.56.0

### PR DESCRIPTION
**Pull request by Allex1**
Addresses https://github.com/open-telemetry/opentelemetry-helm-charts/issues/296

### CODE REVIEW

**Comment by Allex1** [comment]
@TylerHelmuth not sure how to debug this:

```logger.go:42: 10:18:51 | autoscale | 2022-07-30 10:16:25 +0000 UTC	Warning	HorizontalPodAutoscaler.autoscaling simplest-collector		FailedComputeMetricsReplicas	invalid metrics (1 invalid out of 1), first error is: failed to get cpu utilization: unable to get metrics for resource cpu: unable to fetch metrics from resource metrics API: the server could not find the requested resource (get pods.metrics.k8s.io)```

**Comment by TylerHelmuth** [comment]
@Allex1 I think that test on the operator side is kinda flakey

**Review by TylerHelmuth — COMMENTED** [review]
@Allex1 normally the CRDs don't have this many changes, did something big happen over in the operator space? Normally only the additions are added to the files, not copy and paste the whole thing.

**Comment by TylerHelmuth** [comment]
> @Allex1 I think that test on the operator side is kinda flakey

I reran the tests 2 times but it still failed. Id recommend looking at the change history for those tests in the operator repo to see if something changed.

**Comment by Allex1** [comment]
> @Allex1 normally the CRDs don't have this many changes, did something big happen over in the operator space? Normally only the additions are added to the files, not copy and paste the whole thing.

@TylerHelmuth  Actually there are very many differences between 0.54.0 and 0.56.0 crds in the opentelemetry repo. Mostly indentation. 

`git diff v0.56.0 v0.54.0 config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml`

On my last pr upgrading from 0.54.0 to 0.55.0 I only checked the diff on the operator repo (and saw no diff) and not compared to what was in this master ...

**Comment by TylerHelmuth** [comment]
Would be good if we could add a make target in this repo that would automatically update the CRDs. Then we could run a test in the CI similar to Check Examples that would verify things.

**Comment by TylerHelmuth** [comment]
@Allex1 my guess is the failing tests are because of this PR: https://github.com/open-telemetry/opentelemetry-operator/pull/984.  Looks like a new e2e test was added that has a dependency on a "metrics-server".  Our CI doesn't know about it and will need updated to install it: https://github.com/open-telemetry/opentelemetry-operator/blob/main/hack/install-metrics-server.sh

**Comment by Allex1** [comment]
@TylerHelmuth @dmitryax PTAL

**Review by TylerHelmuth — APPROVED** [review]

File: .github/workflows/operator-test.yaml [hunk 1]
```diff
@@ -47,6 +47,12 @@ jobs:
           sudo curl -Lo /usr/local/bin/kubectl-kuttl https://github.com/kudobuilder/kuttl/releases/download/v0.11.1/kubectl-kuttl_0.11.1_linux_x86_64
           sudo chmod +x /usr/local/bin/kubectl-kuttl
 
+      - name: Install metrics-server
+        run: |
+          kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml
+          kubectl patch deployment -n kube-system metrics-server --type "json" -p '[{"op": "add", "path": "/spec/template/spec/containers/0/args/-", "value": --kubelet-insecure-tls}]'
+          kubectl wait --for=condition=available deployment/metrics-server -n kube-system  --timeout=5m
+
```
Comment (dmitryax) [inline .github/workflows/operator-test.yaml:55]: Why is this needed for upgrade?

Comment (TylerHelmuth) [reply .github/workflows/operator-test.yaml:55]: the 0.56.0 Tag of the operator includes new e2e tests that have a dependency on this metrics server.  These commands are copied from https://github.com/open-telemetry/opentelemetry-operator/blob/main/hack/install-metrics-server.sh

Comment (dmitryax) [reply .github/workflows/operator-test.yaml:55]: Ok, sounds good. Thanks for clarifying

**Review by dmitryax — APPROVED** [review]
